test/cpp/microbenchmarks/callback_streaming_ping_pong.h

```diff
@@ -0,0 +1,154 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef TEST_CPP_MICROBENCHMARKS_CALLBACK_STREAMING_PING_PONG_H
+#define TEST_CPP_MICROBENCHMARKS_CALLBACK_STREAMING_PING_PONG_H
+
+#include <benchmark/benchmark.h>
+#include <sstream>
+#include "src/core/lib/profiling/timers.h"
+#include "src/proto/grpc/testing/echo.grpc.pb.h"
+#include "test/cpp/microbenchmarks/callback_test_service.h"
+#include "test/cpp/microbenchmarks/fullstack_context_mutators.h"
+#include "test/cpp/microbenchmarks/fullstack_fixtures.h"
+
+namespace grpc {
+namespace testing {
+
+/*******************************************************************************
+ * BENCHMARKING KERNELS
+ */
+
+class BidiClient
+    : public grpc::experimental::ClientBidiReactor<EchoRequest, EchoResponse> {
+ public:
+  BidiClient(benchmark::State* state, EchoTestService::Stub* stub,
+             ClientContext* cli_ctx, EchoRequest* request,
+             EchoResponse* response)
+      : state_{state},
+        stub_{stub},
+        cli_ctx_{cli_ctx},
+        request_{request},
+        response_{response} {
+    msgs_size_ = state->range(0);
+    msgs_to_send_ = state->range(1);
+    StartNewRpc();
+  }
+
+  void OnReadDone(bool ok) override {
+    if (!ok) {
+      gpr_log(GPR_ERROR, "Client read failed");
+      return;
+    }
+    if (writes_complete_ < msgs_to_send_) {
+      MaybeWrite();
```

Comment:
MaybeWrite does the same check as in the previous line, so why is this code doing the check and then calling MaybeWrite?

Reply:
Yeah, that's an error. Thanks for pointing it out.